File: Python/easy/triangle/triangle.py
def equilateral(sides):
    """This function chechs if a triangle is equilateral

    :param sides: int[] - the size of each side

    Return True if the triangle is equilateral and False otherwise  
    """

    if sides[0] == 0 or sides[1] == 0 or sides[2] == 0:
        return False
    return sides[0] == sides[1] == sides[2]


def isosceles(sides):
    """This function chechs if a triangle is isosceles

    :param sides: int[] - the size of each side

    Return True if the triangle is isosceles and False otherwise  
    """

    if sides[0] + sides[1] < sides[2] or sides[0] + sides[2] < sides[1] or sides[1] + sides[2] < sides[0]:
        return False
    return sides[0] == sides[1] or sides[0] == sides[2] or sides[1] == sides[2]


def scalene(sides):
    """This function chechs if a triangle is scalene

    :param sides: int[] - the size of each side

    Return True if the triangle is scalene and False otherwise  
    """

    if sides[0] + sides[1] < sides[2] or sides[0] + sides[2] < sides[1] or sides[1] + sides[2] < sides[0]:
        return False
    return sides[0] != sides[1] != sides[2] != sides[0]

File: C++/easy/triangle/triangle.cpp
#include "triangle.h"

namespace triangle {
  flavor kind(const double& a, const double& b, const double& c) {
    flavor result{};
    if ((a <= 0.00 || b <= 0.00 || c <= 0.00) ||
       ((a + b < c) || (b + c < a) || (b + c < a)))
      throw std::domain_error("A domain error has occurred");

    if (a == b && b == c) result = flavor::equilateral;
    else if ((a == b && b != c) || (a != b && b == c) || (a == c && c != b))
      result = flavor::isosceles;
    else result = flavor::scalene;
    return result;
  }
}  // namespace triangle
